xarray/core/combine.py

```diff
@@ -207,14 +207,14 @@ def _combine_all_along_first_dim(
 
     # Group into lines of datasets which must be combined along dim
     # need to sort by _new_tile_id first for groupby to work
-    # TODO remove all these sorted OrderedDicts once python >= 3.6 only
-    combined_ids = OrderedDict(sorted(combined_ids.items(), key=_new_tile_id))
+    # TODO: ask for help here
+    combined_ids = dict(sorted(combined_ids.items(), key=_new_tile_id))
```

Comment:
Do we need this `sorted` call here still? Its possible this line can be removed.

Reply:
I think we still need to canonicalize the order of variables between different datasets. @TomNicholas does that sound right to you?

Reply:
I think we can leave this for now.

Reply:
Yes you need the `sorted` call. 

Each object to be combined is assigned a "tile id" which specifies its position relative to the other objects. Sorted lines them all up in the correct order before the concatenation happens, which is [required](https://docs.python.org/3/library/itertools.html#itertools.groupby) by `itertools.groupby`. The `OrderedDict` was originally needed otherwise in python 3.5 this order was not guaranteed to be preserved before concatenation.

There isn't anything left to do here now, the Todo can be removed.